projects/plugins/crm/src/js/state/automations-admin/reducer.ts

```diff
@@ -0,0 +1,60 @@
+import { findStep, getIdentifiedWorkflow } from './util';
+import type { AutomationsAction } from 'crm/state/automations-admin/actions';
+import type { IdentifiedWorkflow } from 'crm/state/automations-admin/types';
+
+export type WorkflowState = { [ workflowId: number ]: IdentifiedWorkflow };
+
+export const workflows = (
+	state: WorkflowState = {},
+	action: AutomationsAction
+): WorkflowState => {
+	switch ( action.type ) {
+		case 'HYDRATE_WORKFLOWS':
+			if ( ! action.workflows ) {
+				return state;
+			}
+			return action.workflows.reduce(
+				( newState, workflow ) => ( {
+					...newState,
+					[ workflow.id ]: getIdentifiedWorkflow( workflow ),
+				} ),
+				{}
+			);
+		case 'ACTIVATE_WORKFLOW':
+			if ( ! action.id || ! state[ action.id ] ) {
+				return state;
+			}
+			return { ...state, ...{ [ action.id ]: { ...state[ action.id ], active: true } } };
+		case 'DEACTIVATE_WORKFLOW':
+			if ( ! action.id || ! state[ action.id ] ) {
+				return state;
+			}
+			return { ...state, ...{ [ action.id ]: { ...state[ action.id ], active: false } } };
```

Comment:
It could actually be simplified to this:
return { ...state, [ action.id ]: { ...state[ action.id ], active: false }, };